cmd/ulxly/bridge_service/legacy/deposit.go

```diff
@@ -0,0 +1,73 @@
+package legacy
+
+import (
+	"math/big"
+	"strconv"
+	"strings"
+
+	"github.com/0xPolygon/polygon-cli/cmd/ulxly/bridge_service"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type GetDepositResponse struct {
+	Deposit DepositResponse `json:"deposit"`
+	Code    *int            `json:"code"`
+	Message *string         `json:"message"`
+}
+
+type DepositResponse struct {
+	LeafType      uint8  `json:"leaf_type"`
+	OrigNet       uint32 `json:"orig_net"`
+	OrigAddr      string `json:"orig_addr"`
+	Amount        string `json:"amount"`
+	DestNet       uint32 `json:"dest_net"`
+	DestAddr      string `json:"dest_addr"`
+	BlockNum      string `json:"block_num"`
+	DepositCnt    uint32 `json:"deposit_cnt"`
+	NetworkID     uint32 `json:"network_id"`
+	TxHash        string `json:"tx_hash"`
+	ClaimTxHash   string `json:"claim_tx_hash"`
+	Metadata      string `json:"metadata"`
+	ReadyForClaim bool   `json:"ready_for_claim"`
+	GlobalIndex   string `json:"global_index"`
+}
+
+func (r *DepositResponse) ToDeposit() (*bridge_service.Deposit, error) {
+	d := &bridge_service.Deposit{}
+	var err error
+	d.BlockNum, err = strconv.ParseUint(r.BlockNum, 10, 64)
+	if err != nil {
+		return nil, err
+	}
+
+	d.GlobalIndex = new(big.Int)
+	d.GlobalIndex.SetString(r.GlobalIndex, 10)
```

Comment:
It could be worth checking the boolean response for success and returning an error if it's not successful.